transformer_lens/HookedTransformer.py

```diff
@@ -406,7 +406,8 @@ def to_tokens(
             return_tensors = "pt", 
             padding = True,
             truncation = truncate,
-            max_length = self.cfg.n_ctx if truncate else None
+            max_length = self.cfg.n_ctx if truncate else None,
+            add_special_tokens = False  # As we manually add the BOS token
```

Comment:
I'm uncomfortable with this applying to _all_ tokenizers in case it results in unexpected bugs, can you replace this with eg `True if self.cfg.original_architecture == "OPTForCausalLM" else False`? (Irritatingly wordy, but this is what `original_architecture` is intended for, and this feels neater than an entire if statement copying the tokenizer run. I think the True, False framing is more readable, even if it is literally equivalent to just `self.cfg.original_architecture == "OPTForCausalLM"`)

Reply:
`self.tokenizer.name` to OPT actually might be more reasonable, since it allows other archs with the same tokenizer (if this would ever happen, idk)